### ISSUE TITLE
More infos about -simple 1 transistor amplifier-

### ISSUE THREAD

**Issue by whallmann** [RPiks/pico-WSPR-tx#12]
This is not really an issue more a question. Before trying this software, are there any informations about the mWatts that falls out of the PI? There was mentioned a simple transistor amplifier.  Is there someone who tries a special one, maybe a scribble of the suroundings? 
Thanks, Wolf DF7PN

**Comment by Jochen-bit** [comment]
Roman hardcoded the WSPR message for output of the pico beacon with 12 dBm (=16mW). 
However, I measured almost 200 mW on a 50 ohm antenna. 
Personally, I think it's cool that the pico beacon does not have a PA. Of course, a good antenna is a prerequisite. 
However, there are many simple circuits on the web with single-transitor power amplifiers for digital signals. I myself recently reached the whole of Europe in the 40m band with a pico using my shack antenna.
Give it a try, the hardware you need is less than 5€ ;-) ... and have fun trying it!
Jochen, DG7JH

**Comment by bradshawlupton** [comment]
> please explain how you measured the power on the pico.  I would like to duplicate your measurements on my device on a 17m dipole that makes it from Boston on the US east coast, all across the country to the west, and multiple spots in europe each day.


I am very pleasantly surprised.   Bradshaw K1TE


Sent with Proton Mail secure email.

**Comment by Jochen-bit** [comment]
I used an oscilloscope to measure the output voltage (6.5Vss).
On the antenna I measured reactance at 18.1 MHz (54 ohms).
From this I calculated the power: 
3.25V²/54Ohm=196mW 
Was that right ;-)?

**Comment by bradshawlupton** [comment]
I am a retired physics and chemistry teacher. and a ham since I was 17.  Dont know, but I will try the same.  Makes sense.  I appreciate the suggestions!


Sent with Proton Mail secure email.

**Comment by bradshawlupton** [comment]
the gpio pins are at 3.3v so I think you are about douple the possible power.  I think it is pwm which is also a power hit.  I use a couple of bright 3.3 while LED bulbs, then "eyeball' it the 3.3v to gnd led is full brightness and the 18.1 mhz one is about half.  Of course I dont know the power output of the led...

**Comment by EngineerGuy314** [comment]
>>I used an oscilloscope to measure the output voltage (6.5Vss).
as others have mentioned, that is unlikely since the pico IO is 3.3V max. 

Also, the IO is really not suited to driving a 50ohm load at RF frequencies, so you won't get anything close to a 3.3v squarewave. I measured about 1volt RMS on the output into 50 ohms using a digital scope. The waveform wasn't a squarewave, but it was also far from a clean sinewave.

I estimate about 20mW power can be delivered (if using two outputs anti-phase). Out of that maybe 10mW of it will be on frequency, the rest will be harmonics.

**Comment by Jochen-bit** [comment]
Yes, you're right. I measured the difference in current consumption (tx on - tx off). This was 13 mA. I don't know what I did wrong. Probably the output was not under load. I also think the transmission power will be between 10 and 20 mW.

**Comment by kholia** [comment]
https://github.com/kholia/Pico-FT8-TX/tree/rig_ctrl?tab=readme-ov-file#amplifier describes the various amplification options.

**Comment by Nothing-Burger-Left-Pizza** [comment]
> I used an oscilloscope to measure the output voltage (6.5Vss). On the antenna I measured reactance at 18.1 MHz (54 ohms). From this I calculated the power: 3.25V²/54Ohm=196mW Was that right ;-)?

Reactance does not contribute to real power, only wattless, non-productive reactive power. Would need to know the reactance **and** the resistance.

Also, I'm pretty sure your 6.5V measurement is peak-to-peak. RMS voltage is lower. If your antenna is resonant (reactance, X, is zero), power will be V<sup>2</sup>/(8R)

**Comment by rogerclarkmelbourne** [comment]
I measured the voltage when feeding a 40m dipole, that uses a commercial balun and has 1:1 SWR, and the voltage appeared to be 4V peak to peak.

The output waveform looks like a square wave with slow rise time and several overshoots, but assuming its duty average power is the same as a sine wave, then the RMS voltage would be around 2.8V

And assuming the antenna is 50 Ohms, the power would be  just under 160mW, which corresponds roughly to what other people have observed.

One problem with this value is that to generate 160mW, the GPIO pin would need to supply around 50mA, which is 2 x the maximum pin current for the Pico :-(


Anyway. The power level that is being transmitted by the beacon is probably far too low and should be changed to at least 100mW

**Comment by rogerclarkmelbourne** [comment]
Here is a screengrab from my Rigol 300MHz DSO of the beacon connected to a 50 Ohm dummy load via a 100nF capactor

The waveform seems to be 1.8V peak to peak.

I calculated the power and it seems to be 32mW, assuming the RMS voltage to be 0.707... of the peak voltage.

By my calculations, the GPIO pin current is 25mA , which is within the spec for this MCU

<img width="800" height="480" alt="Image" src="https://github.com/user-attachments/assets/e0b53d29-784c-40f2-9372-b81716a61f86" />
